src/firefly/java/edu/caltech/ipac/firefly/server/db/DbDataIngestor.java

```diff
@@ -29,54 +31,92 @@
+    //  Supporting functions
+    //====================================================================
+
+    static FileInfo ingestDuckReadable(FormatUtil.Format format, DbAdapter dbAdapter, String source, DataGroup meta, boolean searchForSpectrum)  throws IOException, DataAccessException {
+        if (dbAdapter instanceof DuckDbAdapter) {
+            return DuckDbReadable.castInto(format, dbAdapter).ingestDataDirectly(source, meta);
+        } else if (format == PARQUET) {
+            throw new DataAccessException("Unsupported format (%s), file: %s".formatted(format, source));
+        } else {
+            DataGroup table = DsvTableIO.parse(new File(source), format);
+            return ingestTable(dbAdapter, table, searchForSpectrum);
+        }
+    }
+
+    static FileInfo ingestVoTable(DbAdapter dbAdapter, String source, DataGroup meta, int tblIdx, boolean searchForSpectrum) throws IOException, DataAccessException {
+        if (dbAdapter instanceof DuckDbAdapter) {
+            VoTableReader.parse(new TableParseHandler.DbIngest(dbAdapter, meta, searchForSpectrum), source, tblIdx);
+            return new FileInfo(dbAdapter.getDbFile());
+        } else {
+            DataGroup table = VoTableReader.voToDataGroups(source, tblIdx)[0];
+            return ingestTable(dbAdapter, table, searchForSpectrum);
+        }
+    }
+
+    static FileInfo ingestIpacTable(DbAdapter dbAdapter, File source, DataGroup meta, boolean searchForSpectrum) throws IOException, DataAccessException {
+        if (dbAdapter instanceof DuckDbAdapter) {
+            IpacTableReader.parseTable(new TableParseHandler.DbIngest(dbAdapter, meta, searchForSpectrum), source);   // only the first table.
+            return new FileInfo(dbAdapter.getDbFile());
+        } else {
+            DataGroup table = IpacTableReader.read(source);
+            return ingestTable(dbAdapter, table, searchForSpectrum);
+        }
     }
 
     static FileInfo ingestFitsTable(TableServerRequest req, DbAdapter dbAdapter, String source, int tableIndex) throws IOException, DataAccessException {
         var table = FITSTableReader.convertFitsToDataGroup(source, req, FITSTableReader.DEFAULT, tableIndex);
-        dbAdapter.ingestData(() -> table, dbAdapter.getDataTable());
-        return new FileInfo(dbAdapter.getDbFile());
+        return ingestTable(dbAdapter, table, false);        //logic is already in the reader
     }
 
+    static FileInfo ingestJsonTable(TableServerRequest req, DbAdapter dbAdapter, File srcFile, boolean searchForSpectrum) throws IOException, DataAccessException {
+        var table = JsonToDataGroup.parse(srcFile, req);
+        return ingestTable(dbAdapter, table, searchForSpectrum);
+    }
 
+    static FileInfo ingestTable(DbAdapter dbAdapter, DataGroup table, boolean searchForSpectrum) throws DataAccessException {
+        if (searchForSpectrum) SpectrumMetaInspector.searchForSpectrum(table,true);
```

Comment:
Is the DataGroup only partially read here? Is the meta data all in place or searchForSpectrum with not work.

My bigger question - are there cases where searchForSpectrum was being call before but is not longer being called because of the streaming ingestion.